module/VuFind/src/VuFind/ServiceManager/Factory/AutowiringFactory.php

```diff
@@ -0,0 +1,151 @@
+        ?array $options = null
+    ) {
+        if (!empty($options)) {
+            throw new \Exception('Unexpected options passed to factory.');
+        }
+
+        $reflectionClass = new ReflectionClass($requestedName);
+
+        // Just create the object if there is no constructor:
+        if (null === ($constructor = $reflectionClass->getConstructor())) {
+            return new $requestedName();
+        }
+
+        // Just create the object if there are no constructor parameters:
+        $reflectionParameters = $constructor->getParameters();
+        if (empty($reflectionParameters)) {
+            return new $requestedName();
+        }
+
+        // Map constructor parameters:
+        $params = [];
+        foreach ($reflectionParameters as $reflectionParameter) {
+            $attributes = $reflectionParameter->getAttributes(Autowire::class);
+            $autowireArgs = ($attributes[0] ?? null)?->getArguments();
+            if ($config = $autowireArgs['config'] ?? null) {
+                $params[] = $this->getConfigArray($container, $config);
+            } else {
+                $params[] = $this->resolveService($container, $autowireArgs, $reflectionParameter);
+            }
+        }
+        return new $requestedName(...$params);
+    }
+
+    /**
+     * Get a configuration as an array.
+     *
+     * @param ContainerInterface $container Service container
+     * @param string             $config    Configuration name
+     *
+     * @return array
+     */
+    protected function getConfigArray(ContainerInterface $container, string $config): array
+    {
+        $this->configManager ??= $container->get(ConfigManager::class);
+        return $this->configManager->getConfigArray($config);
+    }
+
+    /**
+     * Resolve service for a constructor parameter.
+     *
+     * @param ContainerInterface  $container           Service container
+     * @param ?array              $autowireArgs        Autowire attribute arguments
+     * @param ReflectionParameter $reflectionParameter Parameter
+     *
+     * @return mixed
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    protected function resolveService(
+        ContainerInterface $container,
+        ?array $autowireArgs,
+        ReflectionParameter $reflectionParameter
+    ) {
+        $name = $autowireArgs['service'] ?? null;
+        if (null === $name) {
+            $type = $reflectionParameter->getType();
+            $name = $type->getName();
+            if (null === $name || !($type instanceof ReflectionNamedType)) {
+                throw new LogicException('Unable to resolve type of parameter ' . $reflectionParameter->getName());
+            }
+        }
+        $containerToUse = ($containerName = $autowireArgs['container'] ?? null)
+            ? $container->get($containerName)
+            : $container;
+        return ($containerToUse instanceof PhpRenderer)
+            ? $containerToUse->plugin((string)$name)
```

Comment:
I suppose this is harmless, but why would we use the renderer as a container when we could instead use the ViewHelperManager?